lib/compat/wordpress-6.8/class-gutenberg-rest-post-counts-controller.php

```diff
@@ -0,0 +1,186 @@
+			return new WP_Error(
+				'rest_cannot_read',
+				__( 'Sorry, you are not allowed to read post counts for this post type.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves post counts for a specific post type.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$post_type = $request['post_type'];
+		$counts    = wp_count_posts( $post_type );
+		$data      = $this->prepare_item_for_response( $counts, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares post counts for response.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param object          $item    Post counts data.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+		foreach ( $fields as $field ) {
+			if ( property_exists( $item, $field ) ) {
+				$data[ $field ] = (int) $item->$field;
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filters post type counts data for the REST API.
+		 * Allows modification of the post type counts data right before it is returned.
+		 *
+		 * @since 6.8.0
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object           $item     The original post counts object.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_post_counts', $response, $item, $request );
+	}
+
+	/**
+	 * Retrieves the post counts schema, conforming to JSON Schema.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		/*
+		 * The fields comprise all non-internal post stati,
+		 * including any custom statuses that may be registered.
+		 * 'trash' is an exception, so if it exists, it is added separately.
+		 */
+		$post_statuses = get_post_stati( array( 'internal' => false ) );
+
+		if ( get_post_status_object( 'trash' ) ) {
+			$post_statuses[] = 'trash';
+		}
+		$schema_properties = array();
+		foreach ( $post_statuses as $post_status ) {
+			$schema_properties[ $post_status ] = array(
+				// translators: %s: Post status.
+				'description' => sprintf( __( 'The number of posts with the status %s.' ), $post_status ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit', 'embed' ),
```

Comment:
Allows embedding in, for example, `WP_Posts_Controller`, e.g., 

```php
		$links['counts'] = array(
			'href'       => rest_url( 'wp/v2/counts/' . $post->post_type ),
			'embeddable' => true,
		);
```

Output of `/wp/v2/posts?_embed=counts`
```json
[
   {
	....,
	_embedded: {
	    "counts": [
	        {
	            "publish": 1,
	            "future": 0,
	            "draft": 0,
	            "pending": 0,
	            "private": 0,
	            "trash": 0
	        }
	},
	...
    ]
},
...
]
```